File: flutter_seller/lib/models/product.dart
class Product {
  final int id;
  final String name;
  final double price;
  final int stock;
  final String category;
  final String imageUrl;
  final double discountPct;

  const Product({
    required this.id,
    required this.name,
    required this.price,
    required this.stock,
    required this.category,
    required this.imageUrl,
    this.discountPct = 0.0,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'] as int,
      name: json['name'] as String,
      price: (json['price'] as num).toDouble(),
      stock: json['stock'] as int,
      category: (json['category'] as String?) ?? 'General',
      imageUrl: (json['image_url'] as String?) ?? '',
      discountPct: (json['discount_pct'] as num?)?.toDouble() ?? 0.0,
    );
  }

  Map<String, dynamic> toJson() => {
        'name': name,
        'price': price,
        'stock': stock,
        'category': category,
        'image_url': imageUrl,
        'discount_pct': discountPct,
      };

  bool get isLowStock => stock <= 10;
}

File: app/models/product.py
"""
product.py — Modelo de Producto en inventario.
"""
from flask import request as flask_request
from ..database.db import db
from datetime import datetime

class Product(db.Model):
    __tablename__ = "products"
    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(120), nullable=False)
    price = db.Column(db.Float, nullable=False)
    stock = db.Column(db.Integer, default=0)
    category = db.Column(db.String(60), default="General")
    image_url = db.Column(db.String(300), default="")
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    sale_items = db.relationship("SaleItem", backref="product", lazy=True)

    def to_dict(self):
        image_url = self.image_url or ""
        if image_url and image_url.startswith("/"):
            try:
                image_url = flask_request.host_url.rstrip("/") + image_url
            except RuntimeError:
                pass  # fuera de contexto de request, dejar relativa
        return {"id": self.id, "name": self.name, "price": self.price,
                "stock": self.stock, "category": self.category,
                "image_url": image_url,
                "created_at": self.created_at.isoformat()}
